packages/client-sdk-nodejs/src/internal/cache-data-client.ts

```diff
@@ -4143,10 +4143,17 @@ export class CacheDataClient implements IDataClient {
     middlewares: Middleware[],
     middlewareRequestContext: MiddlewareRequestHandlerContext
   ): Interceptor[] {
-    return [
+    const lateLoadMiddlewares = middlewares.filter(
+      middleware => middleware.shouldLoadLate && middleware.shouldLoadLate()
+    );
+    const immediateMiddlewares = middlewares.filter(
+      middleware => !middleware.shouldLoadLate
```

Comment:
```suggestion
      middleware => !middleware.shouldLoadLate || (middleware.shouldLoadLate && !middleware.shouldLoadLate())
```

This will keep us from dropping middlewares that have `shouldLoadLate()` defined but have it returning false.

Reply:
Agreed this is kind of confusing. Can we change `shouldLoadLate` to an optional boolean (that by convention defaults to false) instead of an optional boolean function? Then we could write:

```typescript
const immediateMiddlewares = middlewares.filter(middleware => !(middleware.shouldLoadLate ?? false))
```

and similarly for the late loaded middlewares